pay-api/src/pay_api/models/payment.py

```diff
@@ -213,19 +226,24 @@ def search_purchase_history(cls,  # pylint:disable=too-many-arguments, too-many-
 
         if not return_all:
             # Add pagination
-            pagination = query.paginate(per_page=limit, page=page)
-            result, count = pagination.items, pagination.total
+            sub_query = query.with_entities(Invoice.id).\
+                    group_by(Invoice.id).\
+                    limit(limit).\
+                    offset((page - 1) * limit).\
+                    subquery()
+            result, count = query.filter(Invoice.id.in_(sub_query)).all(), query.count()
             # If maximum number of records is provided, return it as total
             if max_no_records > 0:
                 count = max_no_records if max_no_records < count else count
-        else:
+        elif max_no_records > 0:
             # If maximum number of records is provided, set the page with that number
-            if max_no_records > 0:
-                pagination = query.paginate(per_page=max_no_records, page=1)
-                result, count = pagination.items, max_no_records
-            else:
-                result = query.all()
-                count = len(result)
+            sub_query = query.with_entities(Invoice.id).\
+                group_by(Invoice.id).\
+                limit(max_no_records)
+            result, count = query.filter(Invoice.id.in_(sub_query.subquery())).all(), sub_query.count()
```

Comment:
`sub_query.count()` instead of `max_no_records` because it's possible the count could be less than the `max_no_records`